toolkit/tools/imagecustomizerlib/customizeutils.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"io/fs"
+	"path/filepath"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+)
+
+func doCustomizations(baseConfigPath string, config *imagecustomizerapi.SystemConfig, imageChroot *safechroot.Chroot) error {
```

Comment:
Would like to open a discussion on a possibility of Trident calling into this function more directly. Trident will be also setting up chroot during image deployment and could take advantage of this function. We can of course create a more suitable wrapper.

Reply:
Had another comment above, but maybe in the future we can hook into these functions as part of the normal image build flow as well.